src/main/scala/exercises/06_Collections.scala

```diff
@@ -139,23 +153,34 @@ object Collections {
    * As usual, various ways exist: pattern matching, folding, ...
    */
   def maxElementInList(l: List[Int]): Int = {
-    error("fix me")
+    l.foldLeft(0)((m: Int, n: Int) => if (m > n) m else n)
   }
 
   /**
    * Calculate the sum of the equally position elements
    * of the two list
    */
   def sumOfTwo(l1: List[Int], l2: List[Int]): List[Int] = {
-    error("fix me")
+    if (l1.isEmpty) {
+      l2
+    } else if (l2.isEmpty) {
+      l1
+    } else {
+      l1.zip(l2).map({case (m,n) => m + n}) 
+    }
   }
 
   /**
    *  For this exercise preferably make use of the sumOfTwo
    * method above
    */
   def sumOfMany(l: List[Int]*): List[Int] = {
-    error("fix me")
+    var finalList: List[Int] = List.fill(l.length)(0)
+    // Realized I could've done a fold
+    for (arg <- l) {
+      finalList = sumOfTwo(finalList, arg)
+    }
+    finalList
   }
 
   case class Person(age: Int, firstName: String, lastName: String)
```

Comment:
I think we were supposed to rewrite `separateTheYoungFromTheOld` to make it have more functional style?
The imperative version from the given code is 26 lines (I got mine down to two lines and a one line helper function).